src/metadata_validator.py

```diff
@@ -27,18 +27,24 @@ def metadataValidate(configs, job_queue, mongo_dao):
         log.exception(f'Error occurred when initialize metadata validation service: {get_exception_msg()}')
         return 1
     validator = None
-    # activate container protection
-    set_scale_in_protection(True)
 
     #step 3: run validator as a service
+    scale_in_protection_flag = False
     while True:
         try:
             log.info(f'Waiting for jobs on queue: {configs[SQS_NAME]}, '
                             f'{batches_processed} metadata validation(s) have been processed so far')
-            
-            for msg in job_queue.receiveMsgs(VISIBILITY_TIMEOUT):
+
+            msgs = job_queue.receiveMsgs(VISIBILITY_TIMEOUT)
+            if len(msgs) > 0:
+                scale_in_protection_flag = True
+                set_scale_in_protection(scale_in_protection_flag)
```

Comment:
let if else set flag true/false only after that call set_scale_in_protection(scale_in_protection_flag)